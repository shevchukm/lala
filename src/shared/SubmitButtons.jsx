import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'semantic-ui-react';

const SubmitButtons = props =>{
    return(
        <Button.Group>
            <Button onClick={props.onHandleReset}>Reset</Button>
            <Button.Or />
            <Button
                onClick={props.onHandleSubmit}
                positive
            >
                {props.submitName}
            </Button>
        </Button.Group>
    );
};

SubmitButtons.propTypes = {
    onHandleReset: PropTypes.func,
    onHandleSubmit: PropTypes.func,
    submitName: PropTypes.string
};

SubmitButtons.defaultProps = {
    submitName: 'Submit'
}

export default SubmitButtons;
