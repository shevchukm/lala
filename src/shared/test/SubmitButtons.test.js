import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import SubmitButtons from '../SubmitButtons';

configure({ adapter: new Adapter() });

describe('Submit buttons component', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<SubmitButtons />);

        expect(wrapper).toMatchSnapshot();
    });

    it('executes reset function on click', () => {
        const onResetMock = jest.fn();

        const wrapper = shallow(<SubmitButtons onHandleReset={onResetMock} />);

        const resetButton = wrapper.find('Button').first();

        resetButton.simulate('click');

        expect(onResetMock).toHaveBeenCalledTimes(1);
    });

    it('executes submit function on click', () => {
        const onSubmitMock = jest.fn();
        const wrapper = shallow(<SubmitButtons onHandleSubmit={onSubmitMock} />);
        const sumbitButton = wrapper.find('[positive=true]');

        sumbitButton.simulate('click');

        expect(onSubmitMock).toHaveBeenCalledTimes(1);
    });

    it('has submit button with name from props', () => {
        const wrapper = shallow(<SubmitButtons submitName="some test name" />);
        const sumbitButton = wrapper.find('[positive=true]');

        expect(sumbitButton.children().text()).toEqual('some test name');
    });

    it('has submit button with default name if props empy', () => {
        const wrapper = shallow(<SubmitButtons />);
        const sumbitButton = wrapper.find('[positive=true]');

        expect(sumbitButton.children().text()).toEqual('Submit');
    });
});
