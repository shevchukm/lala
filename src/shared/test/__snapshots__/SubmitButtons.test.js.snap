// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Submit buttons component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SubmitButtons
    submitName="Submit"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Button
          as="button"
          role="button"
        >
          Reset
        </Button>,
        <ButtonOr />,
        <Button
          as="button"
          positive={true}
          role="button"
        >
          Submit
        </Button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "as": "button",
          "children": "Reset",
          "onClick": undefined,
          "role": "button",
        },
        "ref": null,
        "rendered": "Reset",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "as": "button",
          "children": "Submit",
          "onClick": undefined,
          "positive": true,
          "role": "button",
        },
        "ref": null,
        "rendered": "Submit",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Button
            as="button"
            role="button"
          >
            Reset
          </Button>,
          <ButtonOr />,
          <Button
            as="button"
            positive={true}
            role="button"
          >
            Submit
          </Button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "as": "button",
            "children": "Reset",
            "onClick": undefined,
            "role": "button",
          },
          "ref": null,
          "rendered": "Reset",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "as": "button",
            "children": "Submit",
            "onClick": undefined,
            "positive": true,
            "role": "button",
          },
          "ref": null,
          "rendered": "Submit",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
